controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStream.java

```diff
@@ -212,10 +224,12 @@ private String getWritersTableName(String id) {
             String metadataTable = getMetadataTableName(id);
             String epochWithTxnTable = getEpochsWithTransactionsTableName(id);
             String writersPositionsTable = getWritersTableName(id);
+            String subscribersTable = getSubscribersTableName(id);
             return CompletableFuture.allOf(storeHelper.createTable(metadataTable),
-                    storeHelper.createTable(epochWithTxnTable), storeHelper.createTable(writersPositionsTable))
-                                    .thenAccept(v -> log.debug("stream {}/{} metadata tables {}, {} & {} created", getScope(), getName(), metadataTable,
-                                            epochWithTxnTable, writersPositionsTable));
+                    storeHelper.createTable(epochWithTxnTable), storeHelper.createTable(writersPositionsTable),
+                    storeHelper.createTable(subscribersTable))
```

Comment:
although this approach is fine, but should we put the list of subscribers in the stream metadata table itself? I mean in principle it would be better if we avoid creating an additional table segment for each stream. 
for example:
key: subscribers value: List<subscriberIds>
key: subscriber-<id> value: <stream-cut>

that way we avoid creating a separate table. and with this model, we dont necessarily need to iterate over entries in a separate table.. instead use two queries - 1 to get list of subscribers and then the streamcut for each subscriber while performing the truncation.

Reply:
Fixed this and added new keys to existing stream metadata table:
K                            --->    V
SUBSCRIBERS        ----> sub1, sub2, sub3.... subn
SUBSCRIBER_sub1 ----> sub1, ts1, StreamCut1
SUBSCRIBER_sub2 -----> sub2, ts2, StreamCut2
. . .